plugins/chat/assets/javascripts/discourse/services/chat-subscriptions-manager.js

```diff
@@ -245,8 +257,18 @@ export default class ChatSubscriptionsManager extends Service {
                   (thread.currentUserMembership.lastReadMessageId || 0) &&
                 !thread.currentUserMembership.isQuiet
               ) {
-                channel.unreadThreadIds.add(busData.thread_id);
+                channel.threadsManager.markThreadUnread(
+                  busData.thread_id,
+                  busData.created_at
+                );
                 thread.tracking.unreadCount++;
+
+                // We have to do this since it means the user is currently looking
+                // at the channel and we don't want the unread indicator to show in
+                // the sidebar (since that is based on the lastViewedAt).
+                if (channel.id === this.chat.activeChannel?.id) {
+                  channel.currentUserMembership.lastViewedAt = new Date();
+                }
```

Comment:
We could avoid this repeated lengthy comment by having a method on currentUserMembership with a clearer name and having the comment only on top of this method definition

Reply:
Yep makes sense, will fix :+1: